src/include/miopen/solver.hpp

```diff
@@ -242,11 +243,26 @@ struct PerformanceConfigConvAsm3x3U : Serializable<PerformanceConfigConvAsm3x3U>
         f(self.output_lines_per_wave, "output_lines_per_wave");
     }
 
+    // clang-format off
+    auto PerfFieldRules() const
+    {
+        return seq::MakeRuleSet(
+            std::make_tuple(seq::Span<int, 0, 9>{}, &PerformanceConfigConvAsm3x3U::limit_wave_cnt),
+            std::make_tuple(seq::Span<int, 1, 8>{}, &PerformanceConfigConvAsm3x3U::filters_per_wave),
+            std::make_tuple(seq::Span<int, 1, 8>{}, &PerformanceConfigConvAsm3x3U::output_lines_per_wave)
+        );
```

Comment:
Why did you have to add this ?

Reply:
What do you mean? This is a declarative way to make an iterable pref config without tons of code duplication for generic search.

Reply:
I see what has happened here. The whole RuleSet concept is basically consteval std::integral_constant based computation.